File: project07/Array_Processing.java
/*
    display values in an array on the screen
    given array, fills it wiht values from keyboard
    
    Haim Hong
    Feb 24
*/

import java.io.*;
import java.util.Scanner;

public class Array_Processing
{
    public static void main (String[] args){
        Scanner keyboard = new Scanner(System.in);


        System.out.println("Enter the size of array");
        int a = keyboard.nextInt(); // get the size
        double array[] = new double[a]; // makes empty array with size from user input
        readArray(array, a); // calls functions

        printArray(array, a);
    }
    /*
    printArray
    Gets array and the size
    outputs the arrays in seperate lines
    */
    public static void printArray(double[] array, int size){
        System.out.println("\nPrinting Arrays");
        for (int i=0; i < size; i++){ // loops
            System.out.println(array[i]); // prints out the array in i part
        }
    }
    /*
    readArray
    Gets array and size
    gets the input from user until the end of the loop
    */
    public static void readArray(double[] array, int size){
        Scanner keyboard = new Scanner(System.in);
        for (int i = 0; i < size; i++){ // loops until its the end of the size
            System.out.println("Enter the value for array: ");
            array[i] = keyboard.nextDouble(); // puts the input into array
        }
    }
}

File: project07/array_processing.rb
#
# Gets input from user of its size and values for array in functions
# outputs the value that's in the array
#
# Haim Hong
# Feb 24
###########3##

# it pritns arrays
def printArray array
    print "\nPrinting Arrays"
    for i in array do # goes through loop to print each value of arrays
        puts i.to_s # in each seperate lines
    end
end

# gets empty array, size
# reads value from user
# stores value into the array
def readArray array, size
    for i in 1..size do # loops until it eaches the size
        puts "Enter the value for array: "
        array[i] = gets.to_i # stores the user input to the array
    end
end

if __FILE__ == $0
    puts "Enter the size of array: "
    size = gets.to_i # gets the user input
    array = Array.new(size) #set the size of new array with user input

    readArray(array, size) # calls read array
    
    printArray(array) # calls print array 
end